crd/overlayextensionconfig/api/v1alpha1/overlayextensionconfig_types.go

```diff
@@ -0,0 +1,57 @@
+//go:build !ignore_uncovered
+// +build !ignore_uncovered
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OverlayExtensionConfig is the Schema for the overlayextensionconfigs API
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:shortName=oec
+// +kubebuilder:printcolumn:name="OverlayExtensionConfig IP range",type=string,priority=1,JSONPath=`.spec.extensionIPRange`
+type OverlayExtensionConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   OverlayExtensionConfigSpec   `json:"spec,omitempty"`
+	Status OverlayExtensionConfigStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// OverlayExtensionConfigList contains a list of OverlayExtensionConfig
+type OverlayExtensionConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []OverlayExtensionConfig `json:"items"`
+}
+
+// OverlayExtensionConfigSpec defines the desired state of OverlayExtensionConfig
+type OverlayExtensionConfigSpec struct {
+	// +kubebuilder:validation:Optional
+	ExtensionIPRange string `json:"extensionIPRange,omitempty"`
+}
+
+type OECState string
+
+const (
+	OECNone      OECState = "OECNone"
```

Comment:
do we default to none? if not when would we ever actually set it to none (needs description

Reply:
Added None as a default value. Log is also [commenting](https://github.com/Azure/azure-container-networking/pull/2892#discussion_r1705850138) on this enum suggesting that we get rid of None and Pending, what do you think?